src/FrankaLightWeightInterface.cpp

```diff
@@ -123,16 +124,27 @@ void FrankaLightWeightInterface::run_controller() {
 }
 
 void FrankaLightWeightInterface::poll_external_command() {
-  if (network::poll(this->command_, this->zmq_subscriber_)) {
+  if (network_interfaces::zmq::receive(this->command_, this->zmq_subscriber_)) {
+    if (this->command_.joint_state.is_empty()) {
+      throw std::runtime_error("Received joint command is empty.");
+    }
     this->last_command_ = std::chrono::steady_clock::now();
+    for (std::size_t i = 1; i < this->command_.control_type.size(); ++i) {
```

Comment:
since this is invoked in a real-time control callback, you can be a bit more efficient (if slightly less readable) as follows (avoid intermediate index variable by using the iterator object directly, and avoid two calls to the `.at` method at each loop)

```c++
this->control_type_ = this->command_.control_type.at(0);
for(auto type_iter = begin(this->command_.control_type) + 1, type_end = end(this->command_.control_type); type_iter != type_end; ++type_iter) {
  if (*type_iter != this->control_type_) {
    throw ...
  }
}
```

Reply:
How about simply:

```cpp
this->control_type_ = this->command_.control_type.at(0);
for(auto& type_iter : this->command_.control_type) {
  if (*type_iter != this->control_type_) {
    throw ...
  }
}
```

At least this is more readable although you do one iteration more than necessary (the first one but that would obviously never fails). Otherwise, a slightly more compact writing:

```cpp
for (auto type_iter = std::next(this->command_.control_type.begin()); type_iter != this->command_.control_type.end(); ++it) {
...
}
```

Reply:
It's worth skipping the first iteration, but I agree std::next is more modern and preferable to `+1`. By adding an `end` variable, I intended not to have to evaluate the `this->command_.control_type.end()` accessor method at each loop step (since the stop condition is evaluated each time) but to be fair the compiler will probably recognise this and do the same thing implicitly. So your second suggestion is fine imo